#!/usr/bin/env python

PACKAGE="fre_row_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       name,                type,   level, description,           default, min, max

gen.add("i_gain",            double_t, 0,  "i gain for turning: ", 0.0,    0.0, 0.5)
gen.add("d_gain",            double_t, 0,  "d gain for turning: ", 0,    0.0, 0.5)
gen.add("p",            double_t, 0,  "p gain for turning: ", 1.0,    0.0, 100.0)

gen.add("use_speed_control", bool_t,   0, "use_speed_control: ",   True)
gen.add("speed_low",         double_t, 0, "minimal drive speed: ", 0.8,     0.0, 20.0)
gen.add("speed_high",        double_t, 0, "max drive speed: ",     1.0,     0.0, 20.0)


gen.add("end_line_meters_threshold", double_t, 0, "end_line_meters_threshold: ", 1.0, 1.0, 5.0)
gen.add("todo_rows", double_t, 0, "rows to do: ", 9.0, 1.0, 200.0)

gen.add("slow_down_obstacle", bool_t,   0, "slow down if obstacle: ",   True)
gen.add("distance_obstacle",         double_t, 0, "accepted distance for obstacle: ", 2.0,     0.8, 20.0)

gen.add("ray_length",        double_t, 0, "ray_length (NOT LESS THAN end_line_meters_threshold): ",          2.0,     0.0, 5.0)
gen.add("max_lateral_rect_dist",        double_t, 0, "max_lateral_rect_dist: ",          1.2,     0.5, 2.0)

gen.add("min_cone_width",    double_t, 0, "min cone width in deg: ", 30.0,  0.0, 90.0)
gen.add("max_cone_width",    double_t, 0, "max cone width in deg: ", 180.0,  0.0, 360.0)

gen.add("multi_cone_enabled", bool_t,  0, "if true, use three cones", True)

gen.add("obstacle_angle",    double_t, 0, "obstacle_angle in deg: ", 2.0,   0.0, 10.0)
gen.add("min_obs_density",   double_t, 0, "min_obs_density: ",     0.001,     0.0, 1.0)
gen.add("sensor_min_range",   double_t, 0, "sensor_min_range: ",     0.8,     0.8, 120.0)

exit(gen.generate(PACKAGE, "crawl_row_param", "CrawlRowParam"))
